src/GovUk.Education.ExploreEducationStatistics.Common/Model/Chart/ChartDataSetConfig.cs

```diff
@@ -0,0 +1,41 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Model.Chart
+{
+    public class ChartDataSetConfig
+    {
+        public ChartDataSetConfigDataSet DataSet;
+        public ChartDataGrouping DataGrouping;
+        public BoundaryLevel BoundaryLevels;
+    }
+
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum ChartDataGroupingType
+    {
+        EqualIntervals,
+        Quantiles,
+        Custom,
+    }
+
+    public class ChartDataSetConfigDataSet
```

Comment:
I'll make the change. I opted for three different `*DataSet` objects as each have different optional parameters - but it seems that might be wrong? Anyway, just so you're aware, with this change now we may have a data set that requires an indicator but `ChartBaseDataSet` will have optional indicators because of how it's used on something else.

Reply:
Good point! Would suggest we merge that into `ChartBaseDataSet` as well, so the final thing should be:


```cs
public class ChartBaseDataSet
{
    public Guid? Indicator;
    public List<Guid> Filters = new();
    public ChartDataSetLocation? Location;
    public string? TimePeriod;
}
```